lib/src/ansi_code.dart

```diff
@@ -0,0 +1,278 @@
+  static const backgroundLightCyan =
+      const AnsiCode._('light cyan', AnsiCodeType.background, 106);
+  static const backgroundWhite =
+      const AnsiCode._('white', AnsiCodeType.background, 107);
+
+  /// The numeric value associated with this code.
+  final int code;
+
+  /// A description of this code.
+  final String name;
+
+  /// The type of code that is represented.
+  final AnsiCodeType type;
+
+  const AnsiCode._(this.name, this.type, this.code);
+
+  /// Represents the value escaped for use in terminal output.
+  String get escape => "\x1B[${code}m";
+
+  /// Wraps [value] with the [escape] value for this code, followed by
+  /// [resetAll].
+  ///
+  /// Returns `value` unchanged if
+  ///   * [value] is `null` or empty
+  ///   * `dart:io` `stdout.supportsAnsiEscapes` is `false`
+  ///   * [type] is [AnsiCodeType.reset]
+  String wrap(String value) => (io.stdout.supportsAnsiEscapes &&
+          type != AnsiCodeType.reset &&
+          value != null &&
+          value.isNotEmpty)
+      ? "$escape$value${resetAll.escape}"
+      : value;
+
+  /// Returns a [String] formatted with the codes provided in [foreground],
+  /// [background], and [style] or [styles].
+  ///
+  /// Returns `value` unchanged if
+  ///   * [value] is `null` or empty
+  ///   * `dart:io` `stdout.supportsAnsiEscapes` is `false`
+  ///   * All of [foreground], [background], [style], or [styles] are `null` or
+  ///     empty.
+  ///
+  /// Throws an [ArgumentError] if by [style] and [styles] are provided.
+  static String wrapWith(String value,
+      {AnsiCode foreground,
+      AnsiCode background,
+      AnsiCode style,
+      Iterable<AnsiCode> styles}) {
+    if (style != null) {
+      if (styles != null) {
+        throw new ArgumentError(
+            "Cannot assign a value to both `style` and `styles`.");
+      }
+      styles = [style];
+    }
+
+    if (!io.stdout.supportsAnsiEscapes || value == null || value.isEmpty) {
+      return value;
+    }
+
+    final codes = <int>[];
+    if (foreground != null) {
+      assert(foreground.type == AnsiCodeType.foreground);
+      codes.add(foreground.code);
+    }
+    if (background != null) {
+      assert(background.type == AnsiCodeType.background);
+      codes.add(background.code);
+    }
+    if (styles != null) {
+      assert(styles.every((ac) => ac.type == AnsiCodeType.style));
+      codes.addAll(styles.toSet().map((ac) => ac.code));
+    }
+
+    if (codes.isEmpty) {
+      return value;
+    }
+
+    codes.sort();
+    return "\x1B[${codes.join(';')}m$value${resetAll.escape}";
+  }
+
+  @override
+  String toString() => "$name ${type.name} ($code)";
+}
+
+const allStyles = const [
+  AnsiCode.styleBold,
+  AnsiCode.styleDim,
+  AnsiCode.styleItalic,
+  AnsiCode.styleUnderlined,
+  AnsiCode.styleBlink,
+  AnsiCode.styleReverse,
+  AnsiCode.styleHidden,
+  AnsiCode.styleCrossedOut
+];
+
+const List<AnsiCode> allForegrounds = const [
```

Comment:
Consider just naming this "foregrounds", "foregroundColors" or "textColors".
The "all" prefix seems superfluous since there are no collections here containing less than all the foreground colors.

Ditto above for styles and below for background colors